src/Workspaces/Core/Portable/Execution/SynchronizationObject.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Serialization;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    /// <summary>
+    /// Base for object that will use ISolutionSynchronizationService framework to synchronize data to remote host
+    /// </summary>
+    internal abstract partial class SynchronizationObject
+    {
```

Comment:
This is pretty confusing name.  Synchronization has a strong connotation in both .net and programming that is around the topic of 'locking.  However, i don't have anything better :) Possibly RemotableData might be better.  Anyone have any ideas?